src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -59,6 +59,29 @@ var HTML = '__html';
 
 var {Namespaces: {html: HTML_NAMESPACE}, getIntrinsicNamespace} = DOMNamespaces;
 
+var NORMALIZE_NEWLINES_REGEX = /\r\n?/g;
+function isServerTextDifferentFromClientText(
+  serverText: string,
+  clientText: string,
+): boolean {
+  if (serverText === clientText) {
+    return false;
+  }
+  // HTML parsing normalizes CR and CRLF to LF.
+  // https://www.w3.org/TR/html5/single-page.html#preprocessing-the-input-stream
+  // If we have a mismatch, it might be caused by that.
+  // We won't be patching up in this case as that matches our past behavior.
+  // TODO: verify that it doesn't have any observable difference.
+  const normalizedClientText = clientText.replace(
+    NORMALIZE_NEWLINES_REGEX,
+    '\n',
+  );
+  if (normalizedClientText === serverText) {
+    return false;
+  }
+  return true;
```

Comment:
This return value is meant to indicate that it *should* patch it up. So this says that we *will* patch up the DOM if it differs, which is not what your comment says.